src/storage-facade.ts

```diff
@@ -45,25 +75,71 @@ const FireStoreStorageProvider: StorageInterface = {
   setAnonymousPortalData: (_portalData: IAnonymousPortalData) => FirebaseImp.setAnonymousPortalData(_portalData),
 
   // Saving and Loading student work
-  watchAnswer:  (embeddableRefId: string, callback: (wrappedAnswer: FirebaseImp.WrappedDBAnswer | null) => void) => FirebaseImp.watchAnswer(embeddableRefId, callback),
-  watchAllAnswers: (callback: (wrappedAnswer: FirebaseImp.WrappedDBAnswer[]) => void) => FirebaseImp.watchAllAnswers(callback),
+  watchAnswer:  (embeddableRefId: string, callback: (wrappedAnswer: IWrappedDBAnswer | null) => void) => FirebaseImp.watchAnswer(embeddableRefId, callback),
+  watchAllAnswers: (callback: (wrappedAnswer: IWrappedDBAnswer[]) => void) => FirebaseImp.watchAllAnswers(callback),
+
+  // Save an answer to Firebase
   createOrUpdateAnswer: (answer: IExportableAnswerMetadata) => FirebaseImp.createOrUpdateAnswer(answer),
   getLearnerPluginStateDocId: (pluginId: number) => FirebaseImp.getLearnerPluginStateDocId(pluginId),
   getCachedLearnerPluginState: (pluginId: number) => FirebaseImp.getCachedLearnerPluginState(pluginId),
   getLearnerPluginState: (pluginId: number) => FirebaseImp.getLearnerPluginState(pluginId),
   setLearnerPluginState: (pluginId: number, state: string) => FirebaseImp.setLearnerPluginState(pluginId,state),
-  checkIfOnline: () => FirebaseImp.checkIfOnline()
-};
+  checkIfOnline: () => FirebaseImp.checkIfOnline(),
 
+  // TODO: Save activity to local JSON file
+  exportActivityToJSON: (activityId?: string) => "" // console.log("Not yet implemented for Firebase Storage")
+};
 
 const indexDBConnection = new DexieStorage();
 
 const DexieStorageProvider = {...FireStoreStorageProvider,
+
   createOrUpdateAnswer: (answer: IExportableAnswerMetadata) => {
-    console.dir(answer);
-    indexDBConnection.answers.put(answer);
-    // TODO: Return the correct type. For now we call through to Firebase
-    return FirebaseImp.createOrUpdateAnswer(answer);
+    const idxDBAnswer = answer as IIndexedDBAnswer;
+    idxDBAnswer.activity = _currentOfflineActivityId;
+    indexDBConnection.answers.put(idxDBAnswer);
+  },
+
+  watchAnswer: (embeddableRefId: string, callback: (wrappedAnswer: IWrappedDBAnswer | null) => void) => {
+    const questionId = refIdToAnswersQuestionId(embeddableRefId);
+    const getAnswerFromIndexDB = (qID: string) => {
+      const foundAnswers = indexDBConnection
+        .answers
+        .where("question_id")
+        .equals(qID).toArray();
+      return foundAnswers.then((answers) => {
+        return answers[0];
+      });
+    };
+
+    getAnswerFromIndexDB(questionId).then( (answer: IIndexedDBAnswer|null) => {
+      if (answer) {
+        callback(docToWrappedAnswer(answer));
+      } else {
+        callback(null);
+      }
+    });
+  },
+
+  exportActivityToJSON: (activityId?: string) => {
```

Comment:
@knowuh / @dstrawberrygirl  - one thing I just thought of - we should add a version string at the top level of this json so that we can import old data if the structure changes in the future.

Reply:
Yup, that is a great idea.